receiver/prometheusreceiver/internal/metrics_adjuster.go

```diff
@@ -129,17 +129,28 @@ func (tsm *timeseriesMap) get(metric pmetric.Metric, kv pcommon.Map) (*timeserie
 	return tsi, ok
 }
 
-// Create a unique timeseries signature consisting of the metric name and label values.
-func getAttributesSignature(kv pcommon.Map) string {
-	labelValues := make([]string, 0, kv.Len())
-	kv.Sort().Range(func(_ string, attrValue pcommon.Value) bool {
+// Create a unique string signature for attribute values sorted by attribute keys.
+func getAttributesSignature(m pcommon.Map) string {
```

Comment:
I don't like the current implementation since it is way too complicated and inefficient. I would simply move "Sort" out to not use the result to unblock your main work, and revisit this in a separate PR.

Reply:
Ok sounds good. Updated to keep Sort here